platform.md

```diff
@@ -276,10 +275,79 @@ All lifecycle phases:
 - MUST read `CNB_PLATFORM_API` from the execution environment and evaluate compatibility before attempting to parse other inputs (see [Platform API Compatibility](#platform-api-compatibility))
 - MUST give command line inputs precedence over other inputs
 
+#### `analyzer`
+The platform MUST execute `analyzer` in the **build environment**
+
+Usage:
+```
+/cnb/lifecycle/analyzer \
+  [-analyzed <analyzed>] \
+  [-daemon] \ # sets <daemon>
+  [-gid <gid>] \
+  [-layers <layers>] \
+  [-log-level <log-level>] \
+  [-order <order>] \
```

Comment:
What's the order used for?

Reply:
I think the intent is for validation purposes. So we can, for instance, fail a build if any of the buildpacks in the provided order request a mixin that the stackpacks can't provide. We could also move up the buildpack API validation to analyzer as well.

Reply:
Oh hmmm... but wouldn't it make sense to do that check during detect? What if a buildpack requests a mixin that is not provided, but is going to fail detect anyway? I feel like it only makes sense to fail if the buildpack detects and is non-optional. Was this outlined in the RFC?

Reply:
Yes, this is captured in the [What it is](https://github.com/buildpacks/rfcs/blob/main/text/0069-stack-buildpacks.md#what-it-is) section of the RFC.

>  Before any lifecycle phases:
>
 >   The platform may compare the list of mixins that are statically required by all buildpacks (in the stacks sections of their buildpack.toml files) with the static list of mixins provided by the stack images and the stack buildpacks (in the stacks sections of their buildpack.toml files), and fail the build if the required mixins cannot be provided.

Reply:
This also matches the way `pack` validates IIRC. They currently validate mixins prior to executing anything, we we are sort of matching. The thought was that we should aggressively validate up front and relax later if there is desire to do so. The opposite is harder to sell since it would be a breaking change.

Reply:
Makes sense, thanks!